test/replication.rest.test.js

```diff
@@ -0,0 +1,484 @@
+        if (err) return done(err);
+
+        LocalUser.updateAll(
+          { id: aliceId },
+          { fullname: 'Alice Smith' },
+          done);
+      });
+    }
+  });
+
+  var USER_PROPS = {
+    id: { type: 'string', id: true }
+  };
+
+  var USER_OPTS = {
+    base: 'User',
+    plural: 'Users', // use the same REST path in all models
+    trackChanges: true,
+    strict: true,
+    persistUndefinedAsNull: true
+  };
+
+  var CAR_PROPS = {
+    id: { type: 'string', id: true, defaultFn: 'guid' },
+    model: { type: 'string', required: true },
+    maker: { type: 'string' }
+  };
+
+  var CAR_OPTS = {
+    base: 'PersistedModel',
+    plural: 'Cars', // use the same REST path in all models
+    trackChanges: true,
+    strict: true,
+    persistUndefinedAsNull: true,
+    acls: [
+      // disable anonymous access
+      {
+        principalType: 'ROLE',
+        principalId: '$everyone',
+        permission: 'DENY'
+      },
+      // allow all authenticated users to read data
+      {
+        principalType: 'ROLE',
+        principalId: '$authenticated',
+        permission: 'ALLOW',
+        accessType: 'READ'
+      },
+      // allow Alice to pull changes
+      {
+        principalType: 'USER',
+        principalId: ALICE.id,
+        permission: 'ALLOW',
+        accessType: 'REPLICATE'
+      },
+      // allow Peter to write data
+      {
+        principalType: 'USER',
+        principalId: PETER.id,
+        permission: 'ALLOW',
+        accessType: 'WRITE'
+      }
+    ]
+  };
+
+  function setupServer(done) {
+    serverApp = loopback();
+    serverApp.enableAuth();
+
+    serverApp.dataSource('db', { connector: 'memory' });
+
+    // Setup a custom access-token model that is not shared
+    // with the client app
+    var ServerToken = loopback.createModel('ServerToken', {}, {
+      base: 'AccessToken',
+      relations: {
+        user: {
+          type: 'belongsTo',
+          model: 'ServerUser',
+          foreignKey: 'userId'
+        }
+      }
+    });
+    serverApp.model(ServerToken, { dataSource: 'db', public: false });
+    serverApp.model(loopback.ACL, { dataSource: 'db', public: false });
+    serverApp.model(loopback.Role, { dataSource: 'db', public: false });
+    serverApp.model(loopback.RoleMapping, { dataSource: 'db', public: false });
+
+    ServerUser = loopback.createModel('ServerUser', USER_PROPS, USER_OPTS);
+    serverApp.model(ServerUser, {
+      dataSource: 'db',
+      public: true,
+      relations: { accessTokens: { model: 'ServerToken' } }
+    });
+
+    ServerCar = loopback.createModel('ServerCar', CAR_PROPS, CAR_OPTS);
+    serverApp.model(ServerCar, {  dataSource: 'db', public: true });
+
+    serverApp.use(function(req, res, next) {
+      debug(req.method + ' ' + req.path);
+      next();
+    });
+    serverApp.use(loopback.token({ model: ServerToken }));
+    serverApp.use(loopback.rest());
+
+    serverApp.set('legacyExplorer', false);
```

Comment:
Outside the scope of this PR... but global defaults might be a nice feature to add. Something like `loopback.defaults.legacyExplorer = false`.